doc/source/overview/high_level.md

```diff
@@ -0,0 +1,508 @@
+| TF/Kera     | Classification | Tabular     |
+| Black-box   |                | Image       |
+
+Let the model be given by $f$, and let $p_t$ be the target probability of class $t$. Let $0<\lambda<1$ be a hyperparameter. This method constructs counterfactual instances from an instance $X$ by running gradient descent on a new instance $X'$ to minimize the following loss.
+
+$$L(X', X)= (f_{t}(X') - p_{t})^2 + \lambda L_{1}(X', X)$$
+
+The first term pushes the constructed counterfactual towards the desired class, and the use of the $L_{1}$ norm encourages sparse solutions.
+
+This method requires computing gradients of the loss in the model inputs. If we have access to the model and the gradients are available, this can be done directly. If not, we can use numerical gradients, although this comes at a considerable performance cost.
+
+A problem arises here in that encouraging sparse solutions doesn't necessarily generate interpretable counterfactuals. This happens because the loss doesn't prevent the counterfactual solution from moving off the data distribution. Thus, you will likely get an answer that doesn't look like something that you'd expect to see from the data.
+
+__TODO:__
+- Picture example. Something similar to: https://github.com/interpretml/DiCE
+
+**Pros**
+- Both a black and white box method
+- Doesn't require access to the training dataset
+
+**Cons**
+- Not likely to give human interpretable instances
+- Requires configuration in the choice of $\lambda$
+
+
+#### Contrastive Explanation Method
+
+| Model-types | Task-types     | Data-types  |
+| ----------- | -------------- | ----------- |
+| TF/Kera     | Classification | Tabular     |
+| Black-box   |                | Image       |
+
+
+CEM follows a similar approach to the above but includes three new details. Firstly an elastic net $\beta L_{1} + L_{2}$ regularizer term is added to the loss. This term causes the solutions to be both close to the original instance and sparse.
+
+Secondly, we require that $\delta$ only adds new features rather than takes them away. We need to define what it means for a feature to be present so that the perturbation only works to add and not remove them. In the case of the MNIST dataset, an obvious choice of "present" feature is if the pixel is equal to 1 and absent if it is equal to 0. This is simple in the case of the MNIST data set but more difficult in complex domains such as color images.
+
+Thirdly, by training an optional autoencoder to penalize counter factual instances that deviate from the data distribution. This works by minimizing the reconstruction loss of the autoencoder applied to instances. If a generated instance is unlike anything in the dataset, the autoencoder will struggle to recreate it well, and its loss term will be high. We require two hyperparameters $\beta$ and $\gamma$ to define the following Loss,
+
+$$L(X'|X)= (f_{t}(X') - p_{t})^2 + \beta L_{1}(X', X) + L_{2}(X', X)^2 + \gamma L_{2} (X', AE(X'))^2$$
```

Comment:
Technically this isn't correct as the prediction term is different from regular CFs (compare with PP above).